chain/client/src/view_client.rs

```diff
@@ -256,10 +265,18 @@ impl ViewClientActor {
             .shard_id_to_uid(shard_id, header.epoch_id())
             .map_err(|err| QueryError::InternalError { error_message: err.to_string() })?;
 
+        let block_maybe_gc = self.block_maybe_garbage_collected(&header);
```

Comment:
let's move this check to 272 - so that it triggers only if we see that block is not in DB

Reply:
Addressed in [ced9ad2](https://github.com/near/nearcore/pull/6342/commits/ced9ad28a40bcf25e704ffad304b8a7a2bd588f6). It required some minor refactoring due to `get_chunk_extras` in the outer scope taking `&mut self`.